baseapp/baseapp.go

```diff
@@ -719,39 +726,85 @@ func (app *BaseApp) runTx(mode runTxMode, txBytes []byte) (gInfo sdk.GasInfo, re
 	// Result if any single message fails or does not have a registered Handler.
 	result, err = app.runMsgs(runMsgCtx, msgs, mode)
 
-	if err == nil {
+	// Case 1: the msg errors and the post handler is not set.
+	if err != nil && app.postHandler == nil {
+		return gInfo, nil, anteEvents, priority, err
```

Comment:
This is a break from current behavior where when `err != nil` we return the result. https://github.com/cosmos/cosmos-sdk/blob/v0.47.0-rc2/baseapp/baseapp.go#L753

Reply:
yes that's the motivation for this change, see https://github.com/cosmos/cosmos-sdk/pull/14593#discussion_r1069500273

Reply:
The result should still be returned when the `postHandler` is NOT set.

Reply:
fixed